x/assetprofile/types/message_add_entry.go

```diff
@@ -59,5 +59,14 @@ func (msg *MsgAddEntry) ValidateBasic() error {
 	if err != nil {
 		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid creator address (%s)", err)
 	}
+
+	if msg.Decimals < 6 || msg.Decimals > 18 {
```

Comment:
why do we have this condition ?

Reply:
Hello @amityadav0  because right now everyone could create an asset with infinite decimals